plugins/flytekit-snowflake/flytekitplugins/snowflake/agent.py

```diff
@@ -0,0 +1,157 @@
+import json
+from dataclasses import asdict, dataclass
+from typing import Optional
+
+import grpc
+import snowflake.connector
+from flyteidl.admin.agent_pb2 import (
+    PERMANENT_FAILURE,
+    SUCCEEDED,
+    CreateTaskResponse,
+    DeleteTaskResponse,
+    GetTaskResponse,
+    Resource,
+)
+from snowflake.connector import ProgrammingError
+
+from flytekit import FlyteContextManager, StructuredDataset, logger
+from flytekit.core.type_engine import TypeEngine
+from flytekit.extend.backend.base_agent import AgentBase, AgentRegistry, convert_to_flyte_state
+from flytekit.models import literals
+from flytekit.models.literals import LiteralMap
+from flytekit.models.task import TaskTemplate
+from flytekit.models.types import LiteralType, StructuredDatasetType
+
+TASK_TYPE = "snowflake"
+
+
+@dataclass
+class Metadata:
+    user: str
+    account: str
+    database: str
+    schema: str
+    warehouse: str
+    table: str
+    query_id: str
+
+
+class SnowflakeAgent(AgentBase):
+    def __init__(self):
+        super().__init__(task_type=TASK_TYPE)
+
+    def get_private_key(self):
+        from cryptography.hazmat.backends import default_backend
+        from cryptography.hazmat.primitives import serialization
+
+        import flytekit
+
+        pk_path = flytekit.current_context().secrets.get_secrets_file(TASK_TYPE, "rsa_key.p8")
```

Comment:
sure, we can replace private_key instead of rsa_key.p8 here.
However, since we need to read the path of the private_key and do the process, we need to use `get_secrets_file`